### PR TITLE
Add new songs "love-divine-all-loves-excelling" and "silent-night-holy-night"

### PR DESCRIPTION
**Pull request by zcline91**
Love the project! I wanted to contribute a little, and figured I'd start with a hymn whose tune was already transcribed.

### CODE REVIEW

**Comment by kenakofer** [comment]
Fantastic stuff, I appreciate it!

File: lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly [hunk 1]
```diff
@@ -0,0 +1,55 @@
+\version "2.22.1"
+#(ly:set-option 'relative-includes #t)
+\include "../../lib/hymn-common.ily"
+
+%% TUNE INFO
+\include "../../shared-tunes/hyfrydol-public.ily"
+quarternoteTempo = 120
+
+%% SONG INFO
+title = \titleText "Love Divine, All Loves Excelling"
```
Comment (kenakofer) [inline lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly]: My stylistic preference has been to keep capitalization grammatically correct, both in lyrics and the title. 

Regarding capitalize-every-word titles, I think my preference against is mostly arbitrary (what I grew up with), and in the minority. You could probably convince me this is the better way.

File: lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+verseCount = 4
+tags = "english christian 4part"
+dateAdded = "2022-05-12"
+\include "../../lib/header.ily"
+
+%% LYRICS
+verseA = \lyricmode {
+  \l Love di -- vine, all loves ex -- cel -- ling, Joy of heaven to earth come down,
+  \l Fix in us Thy hum -- ble dwell -- ing, All Thy faith -- ful mer -- cies crown!
+  \l Je -- sus, Thou art all com -- pas -- sion, Pure, un -- bound -- ed love Thou art;
```
Comment (kenakofer) [inline lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly]: ...So regarding capitalization in the lyrics, I think modern hymnals are pretty split, and I think I have a good argument for more "prose-like" capitalization in hymns:

References to God in hymns are usually capitalized, even when metaphorical, and the capitalization helps call attention to that. For example, I think "Joy" = Jesus = God in the above, and that's why it's capitalized. But I could be wrong.

We get all this ambiguity when it coincides with the structural capitalization of the poetic line. I'm guessing "Pure" is just structural capitalization, but it leaves some ambiguity in there.

If you really prefer it this way, I'm not going to force you to change it, though.

File: lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+poet = \smallText "Text: Charles Wesley, 1747"
+typesetter = "Zachary Cline"
+verseCount = 4
+tags = "english christian 4part"
+dateAdded = "2022-05-12"
+\include "../../lib/header.ily"
+
+%% LYRICS
+verseA = \lyricmode {
+  \l Love di -- vine, all loves ex -- cel -- ling, Joy of heaven to earth come down,
```
Comment (kenakofer) [inline lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly]: I prefer "heav'n" for the single syllable form.

File: lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly [hunk 4]
```diff
@@ -0,0 +1,94 @@
+\version "2.22.1"
+#(ly:set-option 'relative-includes #t)
+\include "../../lib/hymn-common.ily"
+
+%% TUNE INFO
+composer = \smallText "Music: Franz Gruber, 1818"
+meter = \smallText "Irregular"
```
Comment (kenakofer) [inline lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly]: You had no way of knowing this, but the web pages won't generate without a tune name in the meter (sometimes it's the same as the title). For this, based on hymnary.org, I suggest:

`meter = \smallText "STILLE NACHT irregular"`

File: lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+hymnBaseMoment = \set Timing.baseMoment = #(ly:make-moment 1/8)
+hymnBeatStructure = \set Timing.beatStructure = 3,3
+quarternoteTempo = 50
+\include "../../lib/global-parts.ily"
+
+%% SONG INFO
+title = \titleText "Silent Night, Holy Night"
+poet = \twoLineSmallText "Text: Joseph Mohr, 1818;" "tr. John Freeman Young, 1863"
+typesetter = "Zachary Cline"
+verseCount = 1
```
Comment (kenakofer) [inline lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly]: 4 :)

File: lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+tags = "english christian christmas 4part"
+dateAdded = "2022-05-13"
+\include "../../lib/header.ily"
+
+%% NOTES
+soprano = {
+  \globalParts
+  \relative g' {
+    f8.( g16) f8 d4. | f8.( g16) f8 d4. | c'4 c8 a4. | bf4 bf8 f4. |
+    g4 g8 bf8.( a16) g8 | f8.\( g16\) f8 d4. | g4 g8 bf8.\( a16\) g8 | f8.\( g16\) f8 d4. |
```
Comment (kenakofer) [inline lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly:29]: Well done with the `\(` 👏🏼

**Review by kenakofer — APPROVED** [review]
If you're happy and prefer everything in its current state, we can just merge these. I think the only followup change I would have to make is adding a tune to silent night. I can run the site generation scripts once merged.

File: lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly (hunk 1 above)
Comment (zcline91) [reply lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly]: It's your project, so you should definitely be the one deciding these things, especially for consistency's sake if that's how the other hymns are formatted. I made the change.

File: lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly (hunk 2 above)
Comment (zcline91) [reply lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly]: I tried to change the capitalization in both hymns to match the style of others, but let me know if there's anything else that needs fixing.

File: lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly (hunk 3 above)
Comment (zcline91) [reply lilypond/songs/love-divine-all-loves-excelling/love-divine-all-loves-excelling.ly]: done!

File: lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly (hunk 4 above)
Comment (zcline91) [reply lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly]: added!

File: lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly (hunk 5 above)
Comment (zcline91) [reply lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly]: Good catch! Fixed!

File: lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly (hunk 6 above)
Comment (zcline91) [reply lilypond/songs/silent-night-holy-night/silent-night-holy-night.ly:29]: Thanks! How to do this was well documented in the project.

**Comment by zcline91** [comment]
Also, sorry for putting both hymns in one pull request. I accidentally forgot to branch. :facepalm:

**Comment by kenakofer** [comment]
Looks great! I'll get the pages generated

**Comment by kenakofer** [comment]
All published now, but it's always possible to push edits if they need to happen.

https://hymnsinger.com/listing/love-divine-all-loves-excelling
https://hymnsinger.com/listing/silent-night-holy-night

Don't be a stranger!

**Pull request merged**